### DISCUSSION TITLE
Importing types into monaco editor

### DISCUSSION THREAD

**Discussion by stoefln** [mobxjs/mobx-state-tree#1888] [Q&A]
I am using [monaco editor](https://microsoft.github.io/monaco-editor/) in my react-application and would like to show code completion suggestions based on the types I created with MST. There is an [example of how you can import typings](https://microsoft.github.io/monaco-editor/playground.html#extending-language-services-configure-javascript-defaults), but I can't seem to figure out how I could use my existing MST models to generate those typings. 
I am sure there is an easy solution, but I guess I just don't know enough about TS and MST.

Any help is very much appreciated!

**Comment by EmilTholin** [comment]
Hi @stoefln!

Maybe you could set [`declaration` to `true` in your `tsconfig.json`](https://www.typescriptlang.org/tsconfig#declaration) file to emit `.d.ts` files of your MST models, which you then could import into monaco?

**Comment by stoefln** [comment]
thanks, @EmilTholin that looks promising! Unfortunately, I have a JS-only project, so I don't even have a tsconfig.json

**Reply by EmilTholin** [reply]
Ah, I see. You might still be able to emit `.d.ts` files from your JavaScript files if you add a `tsconfig.json` and compile it with `tsc`, but I can't say for certain.
